ProjectNetwork.Services/SkillsService.cs

```diff
@@ -12,13 +12,97 @@ public class SkillsService : ISkillsService
 
 
         private readonly ISkillRepository _skillRepository;
+        private readonly IUserSkillRepository _userSkillRepository;
 
-        public SkillsService(ISkillRepository skillRepository) {
+        public SkillsService(ISkillRepository skillRepository, IUserSkillRepository userSkillRepository) {
             this._skillRepository = skillRepository;
+            this._userSkillRepository = userSkillRepository;
         }
         public IEnumerable<Skill> GetAll()
         {
             return this._skillRepository.GetAll();
         }
+        public Skill GetASkill(int id)
+        {
+            return this._skillRepository.Get(id);
+        }
+
+        public List<Skill> GetMySkills(String userId)
+        {
+            var tempSkils = this._userSkillRepository.GetAll(skills => skills.UserId == userId);
+            List<Skill> mySkills = new List<Skill>();
+            foreach (var userSkill in tempSkils)
+            {
+                mySkills.Add(GetASkill(userSkill.SkillId));
+            }
+            return mySkills;
+        }
+
+        
+        public bool PostSkills(string UserId, int[] skills)
+        {
+
+            try
+            {
+               
+                if (skills != null)
+                {
+                    List<UserSkill> newSkills = new List<UserSkill>();
+                    foreach (int x in skills)
+                    {
+                        var cur = new UserSkill();
+                        cur.UserId = UserId;
```

Comment:
Check if the user with the specified ID exists.

Reply:
SkillsController Line 50